cpp/oneapi/dal/backend/primitives/optimizers/line_search_dpc.cpp

```diff
@@ -42,7 +42,8 @@ Float backtracking(sycl::queue queue,
         .wait_and_throw();
     std::int64_t iter_num = 0;
     Float cur_val = 0;
-    while ((iter_num == 0 || cur_val > f0 + c1 * alpha * df0) && iter_num < 100) {
+    constexpr Float eps = std::numeric_limits<Float>::epsilon();
+    while ((iter_num == 0 || cur_val > f0 + c1 * alpha * df0) && alpha > eps) {
```

Comment:
would iter_num condition be completely removed?

Reply:
We still need to track if it is the first iteration or not. We could have supported a bool flag instead but for me there is no difference in that case

Reply:
Also, I'm thinking about returning not only sycl::event but also iter_num as a result, so I think it is fine to leave it like this.